code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/filter/common/AbstractDefaultResourceManagerFilter.java

```diff
@@ -8,7 +8,7 @@
 
 public class AbstractDefaultResourceManagerFilter extends AbstractResourceManagerFilter implements Priority {
 
-    private static final Pattern DNS_NAME_PATTERN = Pattern.compile("^[a-zA-Z0-9](?!.*--)[a-zA-Z0-9-]*[a-zA-Z0-9]$");
+    private static final Pattern DNS_NAME_PATTERN = Pattern.compile("^[a-zA-Z0-9](?!.*--)[a-zA-Z0-9-]*[a-zA-Z0-9]*$");
```

Comment:
@vincent99 (?!{regex}) is the negative lookahead supported by java to match "something not followed by something" pattern